backend/BMT-backend/Application/Services/OrderService.cs

```diff
@@ -126,6 +126,28 @@ public async Task<List<ReportDto>> GetOrderReportsAsync(ReportRequest report)
             return reportList;
         }
 
+        public async Task<List<OrderDetails>> GetOrdersByDateAndStatusAsync(ReportRequest report)
```

Comment:
Considero que este método tiene el problema de no cumplir con "SOLID", específicamente con la "S". Este método, a través de sus parámetros, tiene más de 1 implementación. A través de la inyección de dependencias, ustedes pueden definir una clase con tipo específico y de la cual pueden especializar sus parámetros, para entonces generar el reporte según se necesite.

Este quizá sea un ejemplo algo viejo pero en los comentarios hay valiosas ideas de implementaciones que pueden adaptarse a este problema: https://stackoverflow.com/questions/39174989/how-to-register-multiple-implementations-of-the-same-interface-in-asp-net-core